core/groupby.cpp

```diff
@@ -60,24 +60,32 @@ template <class T> T GroupBy<T>::flatten(const Dim reductionDim) const {
   return out;
 }
 
+template <class T>
+void sum_impl(const VariableProxy &out_data, const T &data_container,
+              const std::vector<Slice> &group, const Dim reductionDim,
+              const MasksConstProxy &masks) {
+  for (const auto &slice : group) {
+    const auto merged_inverted_masks = ~masks_merge(masks, reductionDim, slice);
```

Comment:
Use `data_container.slice(slice).masks()` to get the correct masks here. No need for the `slice` argument to `masks_merge`?